keanu-project/src/main/java/io/improbable/keanu/model/regression/RegressionModelBuilder.java

```diff
@@ -56,16 +59,23 @@ public RegressionModelBuilder withRegularization(RegressionRegularization regula
      *                        This will represent sigmas if no or ridge regularization is used and will represent betas if lasso regularization is used.
      * @return this
      */
-    public RegressionModelBuilder withPriorOnWeights(double[] means, double[] scaleParameters) {
-        RegressionWeights.checkArrayHasCorrectNumberOfFeatures(means, getFeatureCount());
-        RegressionWeights.checkArrayHasCorrectNumberOfFeatures(scaleParameters, getFeatureCount());
-
+    public RegressionModelBuilder withPriorOnWeights(DoubleVertex means, DoubleVertex scaleParameters) {
         this.priorOnWeightsMeans = means;
         this.priorOnWeightsScaleParameters = scaleParameters;
-
         return this;
     }
 
+    public RegressionModelBuilder withPriorOnWeights(DoubleTensor means, DoubleTensor scaleParameters) {
```

Comment:
for symmetry we ought to have a method ` withPriorOnIntercept(DoubleTensor means, DoubleTensor scaleParameters)`